process/block/preprocess/rewardTxPreProcessor.go

```diff
@@ -326,7 +326,8 @@ func (rtp *rewardTxPreprocessor) RequestBlockTransactions(body block.Body) int {
 	for senderShardID, mbsRewardTxHashes := range missingRewardTxsForShards {
 		for _, mbRewardTxHashes := range mbsRewardTxHashes {
 			requestedRewardTxs += len(mbRewardTxHashes.txHashes)
-			rtp.onRequestRewardTx(senderShardID, mbRewardTxHashes.txHashes)
+			// TODO : change epoch
+			rtp.onRequestRewardTx(senderShardID, mbRewardTxHashes.txHashes, 0)
```

Comment:
you have to request with current epoch. Epoch can be missing as paramenter in RequestTransactions as the best quest is to ask with current epoch. It would simplify the code as well. RequestHandler does know about the current epoch. so it should be good.

Reply:
refactored